File: src/my_cpp_pkg/robot_news_station.cpp
#include <chrono>
#include <functional>
#include <memory>
#include <string>

#include "example_interfaces/msg/string.hpp"
#include "rclcpp/rclcpp.hpp"

using namespace std::chrono_literals;

class RobotNewsStationNode : public rclcpp::Node
{
public:
  RobotNewsStationNode()
  : Node("robot_news_station"), robot_name_("ROS 2")
  {
    publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news", 10);
    timer_ = this->create_wall_timer(
      500ms, std::bind(&RobotNewsStationNode::publish_news, this));
    RCLCPP_INFO(this->get_logger(), "Robot news station has started");
  }

private:
  void publish_news()
  {
    auto message = example_interfaces::msg::String();
    message.data = "Breaking news from " + robot_name_ + ": ROS 2 is awesome!";
    publisher_->publish(message);
  }

  std::string robot_name_;
  rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
  rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<RobotNewsStationNode>());
  rclcpp::shutdown();
  return 0;
}

File: src/my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String
 
class RobotnewsstationNode(Node): 
    def __init__(self):
        super().__init__("robot_news_station") 
        self.publisher_= self.create_publisher(String, "robot_news", 10) 
        self.timer_ = self.create_timer(0.5, self.publish_news)
        self.get_logger().info("robot news station has been started.")
    def publish_news(self):
        msg = String()
        msg.data = "hello"
        self.publisher_.publish(msg)

 
def main(args=None):
    rclpy.init(args=args)
    node = RobotnewsstationNode() 
    rclpy.spin(node)
    rclpy.shutdown()
 
 
if __name__ == "__main__":
    main()